keyboards/planck/keymaps/callum/keymap.c

```diff
@@ -1,331 +1,259 @@
-        KC_F8,        KC_F9,        KC_F10,       KC_VOLU,
-
-        KC_POWER,     KC_F11,       KC_F12,       KC_F13,
-        KC_F14,       KC_F15,       KC_F16,       KC_F17,
-        KC_F18,       KC_F19,       KC_F20,       KC_VOLD,
-
-        _______,      XXXXXXX,      KC_HOME,      SC(KC_TAB),
-        C(KC_TAB),    KC_END,       XXXXXXX,      XXXXXXX,
-        XXXXXXX,      COLEMAK,      QWERTY,       _______,
-
-        _______,      _______,      _______,      _______,
-        KC_MPRV,      KC_MUTE,      KC_MPLY,      KC_MNXT,
-        _______,      _______,      _______,      _______
+    [FUNC] = LAYOUT_planck_grid(
+        rset,   f1,   f2,   f3,   f4,   f5,   f6,   f7,   f8,   f9,  f10, volu,
+        powr,  f11,  f12,  f13,  f14,  f15,  f16,  f17,  f18,  f19,  f20, vold,
+        ____, xxxx, xxxx, xxxx, xxxx, xxxx, xxxx, xxxx, xxxx, xxxx, xxxx, ____,
+        ____, ____, ____, ____, prev, mute, play, next, ____, ____, ____, ____
     ),
-
-    /* MOUSE
-     * ,-----------------------------------------------------------------------.
-     * |     |     |CtrL |CtrU |CtrR | B5  |CtrU | ScL |Up   | ScR |     |     |
-     * |-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----|
-     * |     |     | Ac0 | Ac1 | Ac2 | B4  |CtrL |Left |Down |Right|CtrR |     |
-     * |-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----|
-     * |     |     |     |Home | End | B3  |CtrD | ScD | ScU |TabL |TabR |     |
-     * |-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----|
-     * |     |     |     |     |     | B2  | B1  |     |     |     |     |     |
-     * `-----------------------------------------------------------------------'
-     */
-    [_MOUSE] = LAYOUT_planck_grid(
-        _______,      XXXXXXX,      C(KC_LEFT),   C(KC_UP),
-        C(KC_RIGHT),  KC_BTN5,      C(KC_UP),     KC_WH_R,
-        KC_MS_U,      KC_WH_L,      XXXXXXX,      XXXXXXX,
-
-        _______,      XXXXXXX,      KC_ACL0,      KC_ACL1,
-        KC_ACL2,      KC_BTN4,      C(KC_LEFT),   KC_MS_L,
-        KC_MS_D,      KC_MS_R,      C(KC_RIGHT),  XXXXXXX,
-
-        _______,      XXXXXXX,      XXXXXXX,      KC_HOME,
-        KC_END,       KC_BTN3,      C(KC_DOWN),   KC_WH_U,
-        KC_WH_D,      SC(KC_TAB),   C(KC_TAB),    _______,
-
-        _______,      _______,      _______,      _______,
-        _______,      KC_BTN2,      KC_BTN1,      _______,
-        _______,      _______,      _______,      _______
-    )
 };
 
-#ifdef AUDIO_ENABLE
-float colemak_song[][2] = SONG(COLEMAK_SOUND);
-float qwerty_song[][2] = SONG(QWERTY_SOUND);
-#endif
-
-void set_colemak(void) {
-#ifdef AUDIO_ENABLE
-    stop_all_notes();
-    PLAY_SONG(colemak_song);
-#endif
-    set_single_persistent_default_layer(_COLEMAK);
+bool send_string_if_keydown(keyrecord_t *record, const char *s) {
+    if (record->event.pressed) {
+        SEND_STRING(s);
+    }
+    return false;
```

Comment:
In general, it's much better to return true here, as there is some functionality that runs late in the stack, and may cause issues if you use return false.  

Namely, one shot layers are this way, and there may be more.  

Otherwise, it's functionally the same. 

```suggestion
    return true;
```

Reply:
Good to know! I had been working under the (clearly wrong 😬) assumption that if I handled a keycode in `process_record_user` then I should be returning `false`...